MicrosoftAuthenticatorPlugin.cs

```diff
@@ -0,0 +1,93 @@
+//reference System.dll
+//reference System.Core.dll
+using System;
+using System.Linq;
+using System.Net;
+using System.Security.Cryptography;
+using System.Text;
+using MCGalaxy;
+using MCGalaxy.Authentication;
+
+namespace AuthPlugin
+{
+	public sealed class MicrosoftAuthenticationPlugin : Plugin
+	{
+		public override string name { get { return "MicrosoftAuthenticationPlugin"; } }
+		public override string MCGalaxy_Version { get { return "1.9.3.6"; } }
+
+		public override void Load(bool auto)
+		{
+			Authenticator.Current = new MicrosoftFallbackAuthenticator(Authenticator.Current);
+		}
+
+		public override void Unload(bool auto)
+		{
+			Authenticator.Current = ((MicrosoftFallbackAuthenticator)Authenticator.Current).underlyingAuthenticator;
+		}
+	}
+
+    class MicrosoftFallbackAuthenticator : Authenticator
+    {
+		// underlyingAuthenticator is used to allow the plugin to wrap an already wrapped authenticator if there is one.
+		// In addition, the default authenticator is sealed, and so cannot be inherited / extended.
+		public readonly Authenticator underlyingAuthenticator;
+		private readonly string externalIP; //TODO: Move this out.
+
+		public MicrosoftFallbackAuthenticator(Authenticator underlyingAuthenticator)
+        {
+			this.underlyingAuthenticator = underlyingAuthenticator;
+			externalIP = new WebClient().DownloadString("http://icanhazip.com").Replace("\\r\\n", "").Replace("\\n", "").Trim();

```

Comment:
shouldn't we use ipv4? in case if someone's internet connection supports both ipv4 and ipv6?

Reply:
Yes we should, well spotted. I had this in one of my files but apparently missed it here.